### PR TITLE
Refactor common kernels to qcor standard lib

### PR DESCRIPTION
**Pull request by tnguyen-ornl**
Related to https://github.com/ORNL-QCI/qcor/issues/36

- Refactored `shared` folder in QCOR `examples` to the standard lib location.

- Also, support `shots` data persistence to Accelerator Buffer.

- Updated Ctest configs (no need to add `-I` since we are using standard lib headers now.

### CODE REVIEW

File: examples/ftqc_qrt/deuteron.cpp [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#include "xacc.hpp"
```
Comment (amccaskey) [inline examples/ftqc_qrt/deuteron.cpp:1]: should be able to remove this, OptFunction is also typedef'ed in qcor namespace

File: examples/ftqc_qrt/deuteron.cpp [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+int main(int argc, char **argv) {
+  // Allocate 2 qubits
+  auto q = qalloc(2);
+  // Hamiltonian
+  auto H = 5.907 - 2.1433 * X(0) * X(1) - 2.1433 * Y(0) * Y(1) + .21829 * Z(0) -
+           6.125 * Z(1);
+  auto optimizer = createOptimizer("nlopt");
+  xacc::OptFunction f(
```
Comment (amccaskey) [inline examples/ftqc_qrt/deuteron.cpp:20]: should be able to remove xacc namespace

File: examples/ftqc_qrt/run-with-shots.cpp [hunk 3]
```diff
@@ -0,0 +1,31 @@
+#include <qalloc>
```
Comment (amccaskey) [inline examples/ftqc_qrt/run-with-shots.cpp:1]: This is provided with the syntax handler, shouldn't need to include it here

File: examples/ftqc_qrt/run-with-shots.cpp [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  std::cout << "Run H\n";
+  H(q[0]); 
+}
+__qpu__ void cx_gate(qreg q) { 
+  std::cout << "Run CNOT\n";
+  CX(q[0], q[1]);
+}
+
+__qpu__ void bell(qreg q) {
+  using qcor::xasm;
```
Comment (amccaskey) [inline examples/ftqc_qrt/run-with-shots.cpp:15]: xasm is default, should be able to remove this

**Review by amccaskey — CHANGES_REQUESTED** [review]

File: examples/ftqc_qrt/run-with-shots.cpp (hunk 4 above)
Comment (tnguyen-ornl) [reply examples/ftqc_qrt/run-with-shots.cpp]: Fixed

File: examples/ftqc_qrt/run-with-shots.cpp (hunk 3 above)
Comment (tnguyen-ornl) [reply examples/ftqc_qrt/run-with-shots.cpp]: Fixed

File: examples/ftqc_qrt/deuteron.cpp (hunk 2 above)
Comment (tnguyen-ornl) [reply examples/ftqc_qrt/deuteron.cpp]: Fixed

File: examples/ftqc_qrt/deuteron.cpp (hunk 1 above)
Comment (tnguyen-ornl) [reply examples/ftqc_qrt/deuteron.cpp]: Fixed.

**Review by amccaskey — APPROVED** [review]

**Pull request merged**